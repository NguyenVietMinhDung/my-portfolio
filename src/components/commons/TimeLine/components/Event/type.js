export type Props = {
  organization: string,
  role: ?string,
  startDate: string,
  endDate: ?string,
  description: ?string,
};
