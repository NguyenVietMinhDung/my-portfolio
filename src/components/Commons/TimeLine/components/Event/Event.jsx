// @flow
import React, { memo } from 'react';
import CSSModules from 'react-css-modules';
import styles from './Event.scss';
import { Props } from './type';
import utils from '../../../../../utils';

const Event = (props: Props) => {
  const {
    organization, role, startDate, endDate, description,
  } = props;
  const { date: { getDuration } } = utils;
  const duration = getDuration(startDate, endDate);
  return (
    <li
      styleName="event"
      data-aos="fade-up-left"
      data-aos-duration="2000"
    >
      <div styleName="organization">{organization}</div>
      {role && <div styleName="role">{role}</div>}
      <div styleName="duration">
        <i className="lni-calendar" />
        &nbsp;&nbsp;{duration}
      </div>
      {description && <div>{description}</div>}
    </li>
  );
};

export default memo<Props>(CSSModules(Event, styles));
