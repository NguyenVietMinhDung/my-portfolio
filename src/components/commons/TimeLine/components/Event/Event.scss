@import '../../../../../styles/abstracts/variables';
@import '../../../../../styles/abstracts/mixins';

.event {
	padding: 20px 0;

	.organization {
		@include font(inherit, inherit, $black-color, 18px, 500);
		position: relative;

		&::before {
			position: absolute;
			left: -40px;

			width: 14px;
			height: 14px;

			content: '';

			border: 4px solid $opalescent-color;
			border-radius: 100%;
			background-color: $violet-color;
		}
	}

	.role {
		@include font($default-font-family, inherit, $black-color, 16px, 400);
		padding-bottom: 15px;
	}

	.duration {
		@include font($default-font-family, inherit, inherit, 14px, 100);
		padding-bottom: 10px;
	}
}
