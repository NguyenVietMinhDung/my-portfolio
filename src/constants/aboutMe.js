export const TITLE = 'About Me';
export const GREETING = 'Hi There! I\'m Dung Nguyen';
export const SUMMARY = 'I am an ambitious Web\n'
  + 'Application Developer who\n'
  + 'always tries to improve himself\n'
  + 'everyday, a quick learner who\n'
  + 'can absorb new ideas and can\n'
  + 'communicate clearly and\n'
  + 'effectively. Looking for an\n'
  + 'ambitious company which will\n'
  + 'challenge my developer and\n'
  + 'allows me to continue\n'
  + 'to develop my knowledge and\n'
  + 'potential.';
export const basicInfo = [
  {
    iconClassName: 'lni-calendar',
    name: 'Date of birth',
    value: '16 January 1995',
  },
  {
    iconClassName: 'lni-envelope',
    name: 'Email',
    value: 'ngvietminhdung@gmail.com',
  },
  {
    iconClassName: 'lni-phone-handset',
    name: 'Phone',
    value: '(076) 401-2912',
  },
  {
    iconClassName: 'lni-map-marker',
    name: 'Location',
    value: 'Tan Binh District, HCM City',
  },
];
export const skills = [
  {
    itemId: 'skill-1',
    name: 'JavaScript, HTML, CSS',
    value: 80,
  },
  {
    itemId: 'skill-2',
    name: 'ReactJS',
    value: 60,
  },
  {
    itemId: 'skill-3',
    name: 'SCSS',
    value: 60,
  },
  {
    itemId: 'skill-5',
    name: 'Webpack',
    value: 50,
  },
  {
    itemId: 'skill-6',
    name: 'Node.js',
    value: 60,
  },
  {
    itemId: 'skill-7',
    name: 'Git',
    value: 60,
  },
];
