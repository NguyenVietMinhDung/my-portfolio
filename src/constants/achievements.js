export const TITLE = 'Achievements';
export const timeline = {
  iconClassName: 'lni-certificate',
  items: [
    {
      key: 1,
      organization: 'Amazon Web Services (AWS)',
      startDate: 'August 2020',
      endDate: 'August 2023',
      description: 'AWS Certified Cloud Practitioner',
    },
  ],
};
