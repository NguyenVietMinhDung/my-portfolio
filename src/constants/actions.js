export const ACTIVATE_NAVIGATION_ITEM = 'ACTIVATE_NAVIGATION_ITEM';
export const OPEN_NAVIGATION_MENU = 'OPEN_NAVIGATION_MENU';
