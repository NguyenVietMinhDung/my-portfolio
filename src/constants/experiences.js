export const TITLE = 'Experiences';
export const timeline = {
  iconClassName: 'lni-briefcase',
  items: [
    {
      key: 1,
      organization: 'Ubitec',
      role: 'JAVASCRIPT DEVELOPER',
      startDate: 'September 2019',
      endDate: 'May 2020',
    },
    {
      key: 2,
      organization: 'KMS Technology Inc., Vietnam',
      role: 'SOFTWARE ENGINEER',
      startDate: 'November 2017',
      endDate: 'August 2019',
    },
    {
      key: 3,
      organization: 'KMS Technology Inc., Vietnam',
      role: 'FRESHER',
      startDate: 'September 2017',
      endDate: 'November 2017',
      description: 'I had joined KMS as a fresher before graduating. '
      + 'I had been trained to become an official employee for 2 months. '
      + 'Then, I joined one of the biggest project team in the company.',
    },
  ],
};
