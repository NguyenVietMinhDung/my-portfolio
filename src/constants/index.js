import * as actions from './actions';
import * as app from './app';
import * as navbar from './navbar';
import * as header from './header';
import * as footer from './footer';
import * as aboutMe from './aboutMe';
import * as experiences from './experiences';
import * as educations from './educations';
import * as projects from './projects';
import * as achievements from './achievements';

export default {
  actions,
  app,
  navbar,
  header,
  footer,
  aboutMe,
  experiences,
  educations,
  projects,
  achievements,
};
