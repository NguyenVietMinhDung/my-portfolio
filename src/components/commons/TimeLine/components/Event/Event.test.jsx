import React from 'react';
import { shallow } from 'enzyme';
import { timeline as items } from '../../../../../constants/experiences';
import Event from './Event';

const defaultProps = items[0];

describe('<Event />', () => {
  test('renders correctly', () => {
    const component = shallow(<Event {...defaultProps} />);
    expect(component).toMatchSnapshot();
  });
});
