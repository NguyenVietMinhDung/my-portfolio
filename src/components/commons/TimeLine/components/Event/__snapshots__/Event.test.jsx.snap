// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Event /> renders correctly 1`] = `
<li
  className="event"
  data-aos="fade-up-left"
  data-aos-duration="2000"
>
  <div
    className="organization"
  />
  <div
    className="duration"
  >
    <i
      className="lni-calendar"
    />
      
    undefined - Present
  </div>
</li>
`;
